### ISSUE TITLE
Filtering git log to exclude commit messages by regex

### ISSUE THREAD

**Issue by dmaul09** [gitextensions/gitextensions#8176]
I am looking for the correct regex to use in the Filter box in git extensions (v3.3) to exclude all commits containing a certain phrase, i.e., "[Automation]" To remove the visual clutter of commits I don't care about. Is there a way?

**Comment by vbjay** [comment]
Exactly

[Automation]

Or

Automation

?

**Comment by vbjay** [comment]
Provide a sample commit message you would expect to match.

**Comment by vbjay** [comment]
Start with https://regexr.com/5601t and play with it to see what regex
would work.

**Comment by dmaul09** [comment]
![image](https://user-images.githubusercontent.com/46731126/83635985-26dc1100-a573-11ea-8632-15cceeb8216b.png)

I want to see all commits **EXCEPT** those containing the string "[Automation]"

**Comment by vbjay** [comment]
That's what the regex does. Look at the tests I wrote.  Did you follow the
regexr link?

**Comment by dmaul09** [comment]
**Thank you**, yes.  But that regex, when put inside the Filter box of Git Extensions, results in an empty list.

**Comment by vbjay** [comment]
We might need to adjust the options of the Regex object in our code.

You do know you only include what is between the /'s right?


^((?!\[Automation\]).)*$

**Comment by dmaul09** [comment]
Yes, I did know that.  Looks like maybe the negative lookahead option is not understood by the GitExtensions regex object?

**Comment by gerhardol** [comment]
Git uses Posix BRE by default, any regex in regex101 will not work.
https://gist.github.com/CMCDragonkai/6c933f4a7d713ef712145c5eb94a1816

You can use other regex engines https://git-scm.com/docs/git-log#Documentation/git-log.txt---perl-regexp

**Comment by pmiossec** [comment]
@gerhardol Could you give a working example because even with the documentation, I don't succeed.

Personally, to only exclude a pattern, I found  it easier to pass something like that to the filter box `--invert-grep --grep="\[bot\]"`

**Comment by gerhardol** [comment]
Use _Commit message_ , add string "--invert-grep --fixed-strings  --author=\"[bot]\"" (without testing)

**Comment by pmiossec** [comment]
> Use _Commit message_ , add string "--invert-grep --fixed-strings --author="[bot]"" (without testing)

I have made my testing on commit message (it was working well) but didn't know that for author it is no more working since git 2.35 See https://stackoverflow.com/a/70644305/717372

So the solution to exclude an author is to let filtering on "Commit message" in the "filter type" and pass the value `--perl-regexp --author="^(?!.*\[bot\]).*$"`

**Comment by pmiossec** [comment]
> > Use _Commit message_ , add string "--invert-grep --fixed-strings --author="[bot]"" (without testing)
> 
> this does not filter, but ty for the idea.

Yes, as it is described in the link provided, since git v2.35, `--invert-grep` is working only on commit message (so on `--grep=""` parameter) whereas it was working on other parameter --author, committer, ...-- before. **But it could still be used to exclude based on commit message because the syntax is a lot easier!** (and that's even the answer for the original request.

**Comment by mstv** [comment]
Is there anything open yet?
